cranelift/codegen/src/prelude_lower.isle

```diff
@@ -870,6 +870,13 @@
 (decl pure partial preserve_frame_pointers () Unit)
 (extern constructor preserve_frame_pointers preserve_frame_pointers)
 
+;; This definition should be kept up to date with the values defined in
+;; cranelift/codegen/meta/src/shared/settings.rs
+(type StackSwitchModel extern (enum (None) (Basic) (UpdateWindowsTib)))
+
+(decl stack_switch_model (StackSwitchModel) Type)
+(extern extractor infallible stack_switch_model stack_switch_model)
```

Comment:
Rather than making this an extractor from a type, can we make it a partial constructor?

Then the use would look like

```lisp
(lower (stack_switch store_context_ptr load_context_ptr in_payload)
       (if-let (stack_switch_model) (StackSwitchModel.Basic))
       ...)
```

Reply:
Done, but you meant `(if-let (StackSwitchModel.Basic) (stack_switch_model))`, right?

Is there a particular reason for making it a *partial* constructor? Following what's happening for TLS I gave my `StackSwitchModel` enum a `None` variant to indicate that no model was set, but that means that the `stack_switch_model` constructor can be total, or am I missing something?

Reply:
I might be wrong, but I had thought that you can only use partial constructors in `if-let`s. Might be worth removing the `None` variant from `StackSwitchModel`, if so.

Reply:
I tried it, and turning `stack_switch_model` into a total constructor seems to just work.
I don't understand the magic mechanism for configuring what goes into the backend  `Flags` in detail, but from what I can tell it does not allow you to store something logically equivalent to `Option<StackSwitchModel>` in there, so keeping the `None` variant in `StackSwitchModel` itself seems like the way to go.